packages/vite-environment-provider-workerd/src/index.ts

```diff
@@ -409,6 +422,41 @@ function createHotChannel(webSocket: WebSocket): HotChannel {
   };
 }
 
+function getOptionsFromWranglerConfig(configPath: string) {
+  let configOptions: SourcelessWorkerOptions;
+  try {
+    const { workerOptions } = unstable_getMiniflareWorkerOptions(configPath);
+    configOptions = workerOptions;
+  } catch (e) {
+    console.warn(`WARNING: unable to read config file at "${configPath}"`);
+    return {};
+  }
+
+  const {
+    bindings,
+    textBlobBindings,
+    dataBlobBindings,
+    wasmBindings,
+    kvNamespaces,
+    r2Buckets,
+    d1Databases,
+    compatibilityDate,
+    compatibilityFlags,
+  } = configOptions;
```

Comment:
Why do we need to destructure the `configOptions` rather than just return them? Are there options that we are excluding?

Reply:
Using all the options creates some type issues (for options that don't really apply here such as [`outboundService`](https://github.com/cloudflare/workers-sdk/blob/00f340f7c1709db777e80a8ea24d245909ff4486/packages/miniflare/README.md?plain=1#L458))

So we head to tweak the returned options also in our earlier iteration of this: https://github.com/dario-piotrowicz/vite-runtime-5.1-experimentations/blob/389dbdcfa0abd54b70ad8632ebc316548faf952c/packages/vite-runtime-workerd-plugin/src/index.ts#L231-L234

The difference is that there we were removing the incompatible options while here I am actually just picking what we actually support. This feels safer/more robust to me but (as we know what we support and only allow pass that to miniflare) if you find it problematic/too verbose I am happy to go with the other version we had (I don't have a very strong preference here)

Reply:
That's fine. We'll just have to remember to add additional values to the destructuring if they are added in the future.